adapter/src/utils/getBaseUrl.test.js

```diff
@@ -0,0 +1,61 @@
+import { getBaseUrl } from './getBaseUrl.js'
+
+const testOrigin = 'https://debug.dhis2.org'
+// { [testPath]: expectedPath }
+const testPaths = {
+    '/dev/api/apps/simple-app/index.html': '/dev/',
+    '/analytics_dev/dhis-web-maps/plugin.html': '/analytics_dev/',
+    '/dhis-web-line-listing/index.html': '/',
+    '/dhis-web-user-settings/': '/',
+    '/hmis/staging/v41/api/apps/WHO-Data-Quality-App/app.html':
+        '/hmis/staging/v41/',
+}
```

Comment:
Another ambiguous path we might need to consider...

```suggestion
const testPaths = {
    '/dev/api/apps/simple-app/index.html': '/dev/',
    '/analytics_dev/dhis-web-maps/plugin.html': '/analytics_dev/',
    '/dhis-web-line-listing/index.html': '/',
    '/dhis-web-user-settings/': '/',
    '/hmis/staging/v41/api/apps/WHO-Data-Quality-App/app.html':
        '/hmis/staging/v41/',
    '/api/apps/dhis-web-maps': '/api/apps',
}
```

An installed app could have dhis-web-maps as a name.  We should probably block installation of apps with that prefix in core...

Reply:
Hm yeah that's an interesting case -- at `/api/apps/dhis-web-maps/`, we could either be at a weird custom app with the name `'dhis-web-maps'` at a base URL of `/`, OR we could be at the core Maps app in an instance weirdly hosted with the base URL `/api/apps/`... I'm not sure if it's possible to figure out which case is true here hah

Reply:
It could ping the server to see which base URL works 🤷 or we could decide that this will always assume that's a custom app

Reply:
I think we could add enforcement in app hub and core app install logic to disallow custom apps with the dhis-web prefix, hopefully that would be sufficient to disambiguate these cases?  And I don't think we need to cater to apps really wanting that specific URL slug...

Reply:
Cool, added that test case 🙂 👍